src/commonMain/kotlin/scientifik/gdml/solids.kt

```diff
@@ -45,6 +45,79 @@ data class GDMLOrb(
     var r: Number
 ) : GDMLSolid()
 
+@Serializable
+@SerialName("ellipsoid")
+data class GDMLEllipsoid(
+    override var name: String,
+    var ax: Number,
+    var by: Number,
+    var cz: Number,
+    var zcut1: Number = -1000000.0,
+    var zcut2: Number = 1000000.0
+) : GDMLSolid()
+
+@Serializable
+@SerialName("eltube")
+data class GDMLElTube(
+    override var name: String,
+    var dx: Number,
+    var dy: Number,
+    var dz: Number
+) : GDMLSolid()
+
+@Serializable
+@SerialName("elcone")
+data class GDMLElCone(
+    override var name: String,
+    var dx: Number,
+    var dy: Number,
+    var zmax: Number,
+    var zcut: Number
+) : GDMLSolid()
+
+@Serializable
+@SerialName("paraboloid")
+data class GDMLParaboloid(
+    override var name: String,
+    var rlo: Number,
+    var rhi: Number,
+    var dz: Number
+) : GDMLSolid()
+
+@Serializable
+@SerialName("para")
+data class GDMLParallelepiped(
+    override var name: String,
+    var x: Number,
+    var y: Number,
+    var z: Number,
+    var alpha: Number,
+    var theta: Number,
```

Comment:
It seems like there should be a default here

Reply:
what value? something like 0 or 2Pi?

Reply:
Marked as required in schema. Leaving as is.